base_agent/condition.py

```diff
@@ -52,6 +73,33 @@ def check(self):
         return False
 
 
+class TaskStatusCondition(Condition):
+    def __init__(self, agent, task_memid, status="finished"):
+        super().__init__(agent)
+        self.status = status
+        self.task_memid = task_memid
+
+    def check(self):
+        T = None
+        if self.agent.memory.check_memid_exists(self.task_memid, "Tasks"):
```

Comment:
what if the task isn't in the memory ? We should return False here

Reply:
see response to Anurag above- if the task isn't in memory and its a "finished" check, the condition should return True.  if it is not in memory and it is a "running" or "paused" check, the branch will do nothing, and the final return will be False